### DISCUSSION TITLE
Behavior of dark mode and opacity

### DISCUSSION THREAD

**Discussion by dfallman** [tailwindlabs/tailwindcss#13722] [General]
This gets me every time: https://play.tailwindcss.com/TLino7oA8k

(You need to enable/disable dark mode manually to see, on macOS in **System settings** -> **Appearance**)

What's really the reason `border-opacity-x`, `text-opacity-x`, and `bg-opacity-x` only apply to "light mode?" Hence, to have the opacity show up in dark mode as well you need to specifically add `dark:border-opacity-30` and `dark:bg-opacity-30`.

While this seems to add some unnecessary boiler plate, it also adds confusion (well, at least to me!) Wouldn't it be more intuitive if `bg-opacity-30` applied to both light and dark mode, and if you _didn't_ want it in dark mode you'd add `dark:bg-opacity-100`?

The big daddy -- `opacity-x` -- doesn't have this normal/dark split personality behavior, for instance.

**Comment by wongjn** [comment]
You could consider using [the opacity syntax](https://tailwindcss.com/docs/background-color#changing-the-opacity). It is more apparent about what is happening – that the opacity is "attached" to the color instance. The `border-opacity-*`, `bg-opacity-*`, et al classes are deprecated anyway.

You could expand the opacity syntax to use a CSS variable if you want the opacity to apply to both light and dark: https://play.tailwindcss.com/FBRhJGvOBh

**Reply by dfallman** [reply]
No, I can't use that actually. We're using a JavaScript theming engine that inserts the right base color (e.g.border-red-500), but in some circumstances I want to change the opacity. Depreciating this seems like an overall bad idea.

**Reply by wongjn** [reply]
You could consider modifying the JavaScript theming engine to work with the opacity syntax. Alternatively, you could use `!important` with the deprecated color opacity classes: https://play.tailwindcss.com/GShaLvl0Ux

**Reply by dfallman** [reply]
That’s what I’m doing. The main issue here though isn’t the solution, it’s the problem. I just don’t think the opacity should be reset when changing between dark/light mode — to me this 1) doesn’t have anything to do with dark/light mode, and 2) doesn’t feel consistent in that opacity-x for instance doesn’t work this way.
